### ISSUE TITLE
Add back month and year duration constants

### ISSUE THREAD

**Issue by bobhy** [nushell/nushell#9028]
### Describe the bug

There's a lot of history on this issue, but I think the last set of fixes to stabilize internal representation of duration should allow us to do reliable duration date arithmetic and to produce reliable overflow messages to too-large durations (can't remember, > 5 yr?)

And being able to do duration arithmetic over reasonable ranges will allow `touch -d` to rely on Nu `datetime` expressions rather than reinvent the wheel. #8858 

Last PR affecting these constants I can find is #6613.

### How to reproduce

n/a

### Expected behavior

n/a

### Configuration

n/a

**Comment by fdncred** [comment]
you'll have to prove the case you're trying to make. i'd love to add them back, but i bet it's going to have to be better than always having all months be 30 days and all years be 365 day years.

**Comment by bobhy** [comment]
Worst case, we'll add a stdlib tool to do the arithmetic better.    I'd
like to take another look at `+` and `-` code for dates and durations
first.  The function we need is `duration_from(<start_datetime | default
(date now)>)`

**Comment by bobhy** [comment]
I can see my way through by reimplementing a new type duration as a record and adding custom commands to do the various operations involving durations and datetime (the builtin datetime is OK, AFAIK).  It won't be elegant, but I do think it will resolve all the duration calculation issues (and my financial calculation scenarios).

To do it in Rust will involve a lot of things I'm not familiar with, so I hesitate to go that way.  Unless I can find an experienced guide?

Here's the interpreter work I think is necessary.  This is a SWAG!

The problem is the number of (nano)seconds represented by `1month` depends on the date you're basing it on. And, in general, that means doing the calculation at runtime.

1. Define a new parser shape and type for Duration that retains both the quantity and the unit of measure at runtime.   E.g, XDuration(5, days).
  Maybe it's only used for Month, Quarter, Year, Decade, though I don't think we lose much perf by using it for all units of duration.  (Unless someone's currently doing a lot of calculation involving constant datetime and (constant) duration?)  
  I don't think XDuration() needs to support the compound duration returned by `into duration`, i.e "1wk 4day 6hr 7min".  
  I do think it can support fractional numbers of a single unit, e.g XDuration(5.3, days).  We might have to restrict that  units for which this is non-calendar-dependent:, days and smaller.
3. Parse constants like `1month` to new XDuration rather than old (nanoseconds) duration.
4. Add special case handling for arithmetic operators with Datetime and XDuration:
   * datetime plus xduration ==> datetime, xduration plus datetime ==> datetime -- do calendar arithmetic for new datetime
   * datetime minus xduration ==> datetime -- same kind of calendar arithmetic
   * datetime minus datetime ==> xduration -- this one is tricky:   
  caller *must* be able to specify the desired units for the output xduration.  
  Perhaps implement binary operators like `-days`,  `-months`, `-years`, e.g `<datetime> -days <datetime>`?  If so, `<datetime> - <datetime>` returns XDuration(nnn, nanoseconds), as currently.
  Alternatively, disallow datetime minus datetime entirely and implement a specialized builtin: `duration-between <startDatetime> <endDatetime> --convert <units>`
5. fix up `into duration` 
   * add a `--relative-to <datetime>`, which is used for all calendar arithmetic, for same reasons as datetime minus datetime.
   * deals with XDuration instead of Duration
   * change `--convert <units>` switch to return XDuration, not string.  
   * Current `n1 years, n2 months ... ` output of 'into duration' will then be meaningful (currently it is disinformation, despite the disclaimer in help).
   * Nice to have: `"1wk 4day 6hr 7min" | into duration --convert hour` => XDuration(270.124, hour).  Currently, the compount form is not round-trippable.
 5. Other stuff we'll find out along the way.

**Comment by fdncred** [comment]
@bobhy That sounds reasonable. I've outlined similar steps before. I'd just add that, the code we use, under-the-hood (chrono), does this calculation correct. So, it's not really about the math. When the math happens in nushell, it's correct.

It's about how once the calculation is complete, it gets turned into nanoseconds, which in turn, gets turned back into the duration to display, which at this point, now is wrong, exactly as you've outlined.

**Comment by bobhy** [comment]
I'm going to take a crack at doing something along these lines.  I've created a draft PR #9113 which has just a draft spec.  Interested reviewers invited to comment on the spec and then participate in the actual work later on
